src/include/catalog/attribute_handle.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "catalog/catalog.h"
+#include "catalog/catalog_defs.h"
+#include "catalog/catalog_entry.h"
+#include "storage/sql_table.h"
+#include "transaction/transaction_context.h"
+
+namespace terrier::catalog {
+
+class Catalog;
+struct SchemaCol;
+
+/**
+ * A attribute entry represent a row in pg_attribute catalog.
+ */
+class AttributeEntry : public CatalogEntry<col_oid_t> {
+ public:
+  /**
+   * Constructor
+   * @param oid attribute oid
+   * @param sql_table associated with this entry
+   * @param entry a row in pg_attribute that represents this table
+   */
+  AttributeEntry(col_oid_t oid, catalog::SqlTableRW *sql_table, std::vector<type::TransientValue> &&entry)
+      : CatalogEntry(oid, sql_table, std::move(entry)) {}
+};
+
+/**
+ * An attribute handle provides accessors to the pg_attribute catalog.
+ * Each database has it's own pg_attribute catalog.
+ *
+ * Following description verbatim from the Postgres documentation:
+ * The catalog pg_attribute stores information about table columns.
+ * There will be exactly one pg_attribute row for every column in every
+ * table in the database. (There will also be attribute entries for indexes,
+ * and indeed all objects that have pg_class entries.)
+ *
+ * The term attribute is equivalent to column and is used for historical
+ * reasons.
+ */
+class AttributeHandle {
+ public:
+  /**
+   * Construct an attribute handle
+   * @param catalog catalog ptr
+   * @param pg_attribute a pointer to pg_attribute sql table rw helper instance
+   */
+  explicit AttributeHandle(Catalog *catalog, std::shared_ptr<catalog::SqlTableRW> pg_attribute)
```

Comment:
The `catalog` argument is never used in the constructor. And the constructor below is deprecated, which has an extra `table` argument. I'm confused...

Reply:
AttributeHandle changed to address.

Reply:
Removed.